tests/main/parallel-install-classic/task.yaml

```diff
@@ -0,0 +1,88 @@
+summary: Checks parallel installation of classic snaps
+
+# core does not support classic confinement
+systems: [-ubuntu-core-*]
+
+prepare: |
+    # ensure we have no snap user data directories yet
+    rm -rf /home/test/snap
+    rm -rf /root/snap
+    rm -rf /var/snap/test-snapd-classic-confinement
+    rm -rf /var/snap/test-snapd-classic-confinement_foo
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB"/dirs.sh
+    snap pack "$TESTSLIB/snaps/test-snapd-classic-confinement/"
+
+    case "$SPREAD_SYSTEM" in
+        fedora-*|arch-*|centos-*)
+            # although classic snaps do not work out of the box on fedora,
+            # we still want to verify if the basics do work if the user
+            # symlinks /snap to $SNAP_MOUNT_DIR themselves
+            ln -sf "$SNAP_MOUNT_DIR" /snap
+            ;;
+    esac
+
+    snap set system experimental.parallel-instances=true
+
+restore:
+    snap set system experimental.parallel-instances=null
+
+    case "$SPREAD_SYSTEM" in
+        fedora-*|arch-*|centos-*)
+            rm -f /snap
+            ;;
+    esac
+
+execute: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    install_local test-snapd-classic-confinement --classic
+    install_local_as test-snapd-classic-confinement test-snapd-classic-confinement_foo --classic
+
+    su -l -c '! test -d ~/snap/test-snapd-classic-confinement' test
+    su -l -c '! test -d ~/snap/test-snapd-classic-confinement_foo' test
+    not test -d ~/snap/test-snapd-classic-confinement
+    not test -d ~/snap/test-snapd-classic-confinement_foo
+
+    # running the snap as a user works
+    su -l -c "test-snapd-classic-confinement_foo.sh -c 'echo foo'" test | MATCH foo
+    # user directory was created
+    su -l -c 'test -d ~/snap/test-snapd-classic-confinement_foo' test
+
+    # running as root works
+    test-snapd-classic-confinement_foo.sh -c "echo foo" | MATCH foo
+
+    # instance environment variables are correctly set up
+    su -l -c 'test-snapd-classic-confinement_foo.sh -c env' test > snap_foo-env.txt
```

Comment:
Nice

Perhaps this should be move to the `tests/main/snap-env` test?

Reply:
We can, once we drop the experimental flag. Right now, I'd keep this separate in parallel instances tests.